File: superstitioMain/java/superstitio/cards/maso/PowerCard/Sensitive3000.java
package superstitio.cards.maso.PowerCard;

import com.megacrit.cardcrawl.actions.common.GainEnergyAction;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import superstitio.DataManager;
import superstitio.cards.SuperstitioCard;
import superstitio.cards.maso.MasoCard;
import superstitio.powers.EasyBuildAbstractPowerForPowerCard;
import superstitio.powers.SexualHeat;
import superstitio.powers.patchAndInterface.interfaces.orgasm.OnOrgasm_onSquirt;


public class Sensitive3000 extends MasoCard {
    public static final String ID = DataManager.MakeTextID(Sensitive3000.class);

    public static final CardType CARD_TYPE = CardType.POWER;

    public static final CardRarity CARD_RARITY = CardRarity.RARE;

    public static final CardTarget CARD_TARGET = CardTarget.SELF;

    private static final int COST = 2;
    private static final int COST_UPGRADED_NEW = 1;

    private static final int MAGIC = 1;
    private static final int UPGRADE_MAGIC = 0;

    public Sensitive3000() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC);
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        addToBot_applyPower(new JuicyPussy_TempPower(this.magicNumber));
    }

    @Override
    public void upgradeAuto() {
        upgradeBaseCost(COST_UPGRADED_NEW);
    }

    public static class JuicyPussy_TempPower extends EasyBuildAbstractPowerForPowerCard implements OnOrgasm_onSquirt {
        public JuicyPussy_TempPower(int amount) {
            super(amount);
        }

        @Override
        public void updateDescriptionArgs() {
            setDescriptionArgs(amount);
        }

        @Override
        public void onSquirt(SexualHeat SexualHeatPower, AbstractCard card) {
            this.flash();
            this.addToBot(new GainEnergyAction(this.amount));
        }

        @Override
        protected SuperstitioCard makePowerCard() {
            return new Sensitive3000();
        }
    }
}

File: superstitioMain/java/superstitio/cards/maso/PowerCard/Sensitive3000.kt
package superstitio.cards.maso.PowerCard

import com.megacrit.cardcrawl.actions.common.GainEnergyAction
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.monsters.AbstractMonster
import superstitio.DataManager
import superstitio.cards.IsMasoCard
import superstitio.cards.SuperstitioCard
import superstitio.cards.maso.MasoCard
import superstitio.powers.EasyBuildAbstractPowerForPowerCard
import superstitio.powers.SexualHeat
import superstitio.powers.patchAndInterface.interfaces.orgasm.OnOrgasm
import superstitioapi.utils.setDescriptionArgs
@IsMasoCard
class Sensitive3000 : MasoCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET)
{
    init
    {
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC)
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?)
    {
        addToBot_applyPower(JuicyPussy_TempPower(this.magicNumber))
    }

    override fun upgradeAuto()
    {
        upgradeBaseCost(COST_UPGRADED_NEW)
    }

    class JuicyPussy_TempPower(amount: Int) : EasyBuildAbstractPowerForPowerCard(amount), OnOrgasm
    {
        override fun updateDescriptionArgs()
        {
            setDescriptionArgs(amount)
        }

        override fun onOrgasm(SexualHeatPower: SexualHeat)
        {
            this.flash()
            this.addToBot(GainEnergyAction(this.amount))
        }

        override fun makePowerCard(): SuperstitioCard
        {
            return Sensitive3000()
        }
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(Sensitive3000::class.java)

        val CARD_TYPE: CardType = CardType.POWER

        val CARD_RARITY: CardRarity = CardRarity.RARE

        val CARD_TARGET: CardTarget = CardTarget.SELF

        private const val COST = 2
        private const val COST_UPGRADED_NEW = 1

        private const val MAGIC = 1
        private const val UPGRADE_MAGIC = 0
    }
}

